releases/ReleaseCompatibility.md

```diff
@@ -0,0 +1,103 @@
+# Releases & Compatibility
+
+To date, we have been somewhat cavalier about making breaking
+changes in Fairlearn.
+Although we are currently pre-`v1.0.0` and hence without particular
+commitments to compatibility (as generally understood - see e.g.
+the [Semantic Versioning Scheme](https://semver.org/)), we should
+work to reduce the number of breaking changes we make.
+
+## The Problem
+
+At the time of writing, we have just released `v0.4.6`, which has
+some substantial breaking changes in the Metrics area of Fairlearn.
+Before it, `v0.4.5` reworked a smaller subset of this functionality, 
+caused smaller breakages.
+There are also further changes to Metrics planned, which may well
+cause further breaks (although these should be more minor).
+All this is obviously undesirable from a user standpoint - these look
+like 'patch' level releases, but are actually breaking their code.
+Concretely, we've had users install Fairlearn with `pip`, and then
+find themselves unable to run Notebooks from our GitHub project - not
+because the functionality was missing, but because it had been
+renamed.
+Moving our notebooks to being generated as part of the documentation
+[in our examples directory](https://github.com/fairlearn/fairlearn/tree/master/examples)
+will help with this, since this will result in the notebooks being
```

Comment:
Not automatically. I skipped this part because it was making everything A LOT more complicated and I wanted to first get *something* up.

Reply:
Are we still expecting to have the automatic generation in place? I don't mind this doc talking about future things, so long as there's a reasonable prospect of 'future' becoming 'now' within a few weeks.

Reply:
It wasn't on my list for the immediate future since it very much seemed non-trivial. If we want this we should probably spend an hour or two understanding what scikit-learn does for this, and then we should have an idea on how much work it actually is.

Reply:
Actually, I don't think the status of this move matters for this document, now that I have changed it to running the tests rather than the notebooks.